? build
? crl.patch
? lib/ldap.jar
? lib/ldapbp.jar
? lib/ldapsec.jar
Index: src/org/globus/common/CoGProperties.java
===================================================================
RCS file: /home/dsl/cog/CVS/jglobus/src/org/globus/common/CoGProperties.java,v
retrieving revision 1.16
diff -r1.16 CoGProperties.java
62a63,65
>     public static final String CRL_CACHE_LIFETIME =
>         "org.globus.jglobus.crl.cache.lifetime";
> 
510c513
<      * value is the number of seconds the key/pair is cached.
---
>      * value is the number of milliseconds the key/pair is cached.
537a541,576
> 
>     /**
>      * Returns the CRL cache lifetime. If this property is not set or
>      * set to zero or less, no caching is done. The value is the
>      * number of milliseconds the CRLs are cached without checking for
>      * modifications on disk.
>      *
>      * @throws NumberFormatException if the cache lifetime property
>      *         could not be parsed
>      * @return the CRL cache lifetime in milliseconds
>      */
>     public long getCRLCacheLifetime()
>         throws NumberFormatException {
> 
>         long value = 0;
> 
>         String property = getProperty(CRL_CACHE_LIFETIME);
>         if (property != null && property.length() > 0) {
>             long parsedValue  = Long.parseLong(property);
>             if (parsedValue > 0) {
>                 value = parsedValue;
>             }
>         }
> 
>         // System property takes precedence
>         property = System.getProperty(CRL_CACHE_LIFETIME);
>         if (property != null && property.length() > 0) {
>             long parsedValue = Long.parseLong(property);
>             if (parsedValue > 0) {
>                 value = parsedValue;
>             }
>         }
> 
>         return value;
>     }
> 
Index: src/org/globus/gsi/CertificateRevocationLists.java
===================================================================
RCS file: /home/dsl/cog/CVS/jglobus/src/org/globus/gsi/CertificateRevocationLists.java,v
retrieving revision 1.11
diff -r1.11 CertificateRevocationLists.java
227a228,235
>          private final long lifetime;
>          private long lastRefresh;
> 
>          public DefaultCertificateRevocationLists() {
>              lifetime =
>                  CoGProperties.getDefault().getCRLCacheLifetime();
>          }
>  
229c237,241
<             reload(getDefaultCRLLocations());
---
>              long now = System.currentTimeMillis();
>              if (lastRefresh + lifetime <= now) {
> 		 reload(getDefaultCRLLocations());
>                lastRefresh = now;
> 	     }
